modules/standard/Sort.chpl

```diff
@@ -471,38 +498,47 @@ proc sort(ref x: [], comparator:? = new DefaultComparator(),
           param stable:bool = false) {
   chpl_check_comparator(comparator, x.eltType);
 
-  const D = x.domain;
+  if stable {
+    // TODO: implement a stable merge sort with parallel merge
+    // TODO: create an in-place merge sort for the stable+minimizeMemory case
+    // TODO: create a stable variant of the radix sort
+    compilerError("stable sort not yet implemented");
+  } else {
+    unstableSort(x, comparator, x.domain.dim(0));
+  }
+}
 
-  if D.low >= D.high then
-    return;
+/*
+
+Sort the elements in the range 'region' within in the 1D rectangular array
+``x``.  After the call, ``x[region]`` will store elements in sorted order.
+This function accepts a 'region' range as an optimized alternative to using an
+array view.
+
+See the ``proc sort`` declared just above for details.
+
+.. note::
+
+  Due to uncertainty about the usefulness of this routine, it is unstable.
+  Please comment on issue NUMBER TODO if you find this routine important in your
+  work.
+
+ */
+@unstable("sort with a region argument is unstable")
+proc sort(ref x: [], comparator, region: range(?), param stable:bool = false) {
```

Comment:
I did it this way just to prevent potential for problems with overload matching. I agree it is reasonable to relax this in the future, but we can rely on it being unstable when doing so.